### PR TITLE
[DOCS] Fixes adaptive_allocations examples

### PR DESCRIPTION
**Pull request by szabosteve**
## Overview

This PR fixes the API examples that enable adaptive allocations by adding `"num_threads": 1` to the request.

### CODE REVIEW

File: docs/reference/inference/service-elasticsearch.asciidoc [hunk 1]
```diff
@@ -179,6 +179,7 @@ PUT _inference/text_embedding/my-e5-model
       "min_number_of_allocations": 3,
       "max_number_of_allocations": 10
     },
+    "num_threads": 1
```
Comment (jan-elastic) [inline docs/reference/inference/service-elasticsearch.asciidoc:182]:
```suggestion
    "num_threads": 1,
```

File: docs/reference/ml/trained-models/apis/update-trained-model-deployment.asciidoc [hunk 2]
```diff
@@ -119,7 +119,8 @@ POST _ml/trained_models/elastic__distilbert-base-uncased-finetuned-conll03-engli
     "enabled": true,
     "min_number_of_allocations": 3,
     "max_number_of_allocations": 10
-  }
+  },
+  "num_threads": 1
```
Comment (jan-elastic) [inline docs/reference/ml/trained-models/apis/update-trained-model-deployment.asciidoc:123]: The update call shouldn't have `num_threads`. You set the number of threads upon creation and can never change it.

**Review by jan-elastic — CHANGES_REQUESTED** [review]
Reviewed; thanks for keeping the doc in good shape

File: docs/reference/ml/trained-models/apis/update-trained-model-deployment.asciidoc (hunk 2 above)
Comment (szabosteve) [inline docs/reference/ml/trained-models/apis/update-trained-model-deployment.asciidoc:123]:
```suggestion
  }
```

**Review by jan-elastic — APPROVED** [review]
LGTM

**Pull request merged**